libnavigation-base/src/test/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresherTest.kt

```diff
@@ -0,0 +1,304 @@
+package com.mapbox.navigation.base.internal.route
+
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import com.mapbox.api.directions.v5.models.MaxSpeed
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+
+@RunWith(Parameterized::class)
+class AnnotationsRefresherTest(
+    private val oldAnnotation: LegAnnotation?,
+    private val newAnnotation: LegAnnotation?,
+    private val legGeometryIndex: Int,
+    private val expectedMergedAnnotation: LegAnnotation?,
+    private val description: String,
+) {
+
+    companion object {
+
+        @JvmStatic
+        @Parameterized.Parameters(name = "{4}")
+        fun data(): Collection<Array<Any?>> {
```

Comment:
Is there a case when route is so long, that it refreshes partially?

Reply:
What do you mean? I don't have any "route" here, only `LegAnnotation`. It refreshes partially when 0 < index < 5.

Reply:
I mean, what if a single leg in a route is so long, that Directions API returns update only for a part of it?

Reply:
Will there be a different `LegAnnotation` object? If not, I'm not sure it really matters here. How can I even simulate it?

Reply:
@VysotskiVadim yes, if a single leg is very long, it might only receive partial annotations during refresh.